lib/routes/account.js

```diff
@@ -960,7 +970,21 @@ module.exports = function (
         log.begin('Account.deviceDestroy', request)
         var sessionToken = request.auth.credentials
         var uid = sessionToken.uid
-        db.deleteDevice(uid, request.payload.id).then(reply, reply)
+        db.deleteDevice(uid, request.payload.id)
+          .then(reply, reply)
```

Comment:
This is very strange and a little scary.  I can have a go at trying to debug, can you say any more about exactly what symptoms you were observing?